### ISSUE TITLE
pytest-html suppresses terminal "Captured log teardown" output

### ISSUE THREAD

**Issue by alexryabkov** [pytest-dev/pytest-html#446]
"Captured log teardown" section is absent in terminal output if pytest-html plugin is invoked.

pytest-html==3.1.1

Sample code:
```
import pytest
import logging


@pytest.fixture(autouse=True)
def log_wrapper():
    logging.info('- Setup')
    yield
    logging.info('- Teardown')


def test_setup_teardown():
    assert True
```

**Without pytest-html invoked:**
```
$ pytest -rA --log-level=INFO test_setup_teardown.py
...
======================================== PASSES =========================================
__________________________________ test_setup_teardown __________________________________
---------------------------------- Captured log setup -----------------------------------
INFO     root:test_setup_teardown.py:7 - Setup
--------------------------------- Captured log teardown ---------------------------------
INFO     root:test_setup_teardown.py:9 - Teardown
================================ short test summary info ================================
PASSED test_setup_teardown.py::test_setup_teardown
=================================== 1 passed in 0.01s ===================================
```

**With pytest-html invoked:**
```
$ pytest -rA --log-level=INFO --html report.html test_setup_teardown.py
...
======================================== PASSES =========================================
__________________________________ test_setup_teardown __________________________________
---------------------------------- Captured log setup -----------------------------------
INFO     root:test_setup_teardown.py:7 - Setup
--------- generated html file: file:///Users/alex/pytest-issue/report.html ----------
================================ short test summary info ================================
PASSED test_setup_teardown.py::test_setup_teardown
=================================== 1 passed in 0.01s ===================================
```

**Comment by northernSage** [comment]
It seems the culprit is `test_report.when = "call"` in [HTTMLReport._post_process_reports](https://github.com/pytest-dev/pytest-html/blob/master/src/pytest_html/html_report.py#L309). The assignment is setting all runtest "teardown" phases to "call" for the [pytest.reports.TestReport](https://docs.pytest.org/en/6.2.x/reference.html#testreport) objects and pytest isn't outputing "teardown" captures since it's now at "call" phase (just assuming here). 

After doing some tests, setting [pytest.reports.TestReport.when](https://docs.pytest.org/en/6.2.x/reference.html#pytest.reports.TestReport.when) back to "teardown" on every last `TestReport` object in [HttmlReport.reports](https://github.com/pytest-dev/pytest-html/blob/master/src/pytest_html/html_report.py#L34) in the `pytest_sessionfinish` hook after generating and saving the report as follows seem to solve the issue without breaking any tests:

```python
def pytest_sessionfinish(self, session):
    self._post_process_reports()
    report_content = self._generate_report(session)
    self._save_report(report_content)

    # setting reports back to teardown phase
    for r in self.reports.values():
        r[-1].when = "teardown"
```
I would gladly write a PR for this if someone more acquainted with the code base gives me the green light :bowtie:

**Comment by jeffwright13** [comment]
Is there any reason the offered PR has not been accepted yet?

**Comment by BeyondEvil** [comment]
> Is there any reason the offered PR has not been accepted yet?

Yes, we're working on a complete re-write of the plugin - this is taking 100% of our time, so no new PRs are being accepted. We're also looking for people to contribute to and test the re-write. Let me know if you're interested.

**Comment by jeffwright13** [comment]
I would be interested, actually. I am the author of [pytest-tui](https://github.com/jeffwright13/pytest-tui), which also has an HTML report available as output, so i'm somewhat versed in the Pytest code. Maybe we could all put our heads together and Make Pytest Great Again (lol).

**Comment by BeyondEvil** [comment]
> I would be interested, actually. I am the author of [pytest-tui](https://github.com/jeffwright13/pytest-tui), which also has an HTML report available as output, so i'm somewhat versed in the Pytest code. Maybe we could all put our heads together and Make Pytest Great Again (lol).

That would be awesome! Shoot me an email at jim.brannlund+pytest@gmail.com and we can jump on a quick zoom call. The list of what needs to be done is not long, but mostly clear.
